diff --git a/src/server/MinimizationQuery.cpp b/src/server/MinimizationQuery.cpp
index 3a8b824..9471d09 100644
--- a/src/server/MinimizationQuery.cpp
+++ b/src/server/MinimizationQuery.cpp
@@ -5,6 +5,13 @@
  *      Author: dkoes
  */
 
+#ifdef _WIN32
+#define WIN32_LEAN_AND_MEAN
+#include <windows.h>
+#else
+#include <unistd.h>
+#endif
+
 #include <sstream>
 
 #include "MinimizationQuery.h"
@@ -244,7 +251,13 @@ void MinimizationQuery::thread_minimize(MinimizationQuery* q)
 			if (q->stopQuery) //cancelled
 				return;
 			else if (ligands.size() == 0) //nothing to do yet
+			{
+			#ifdef _WIN32
+				Sleep(1);
+			#else
 				usleep(10);
+			#endif
+			}
 			else //let's minimize!
 			{
 				vector<Result*> results;
@@ -280,7 +293,7 @@ void MinimizationQuery::thread_minimize(MinimizationQuery* q)
 				}
 
 				//add computed results
-				boost::lock_guard<shared_mutex> lock(q->results_mutex);
+				boost::lock_guard<boost::shared_mutex> lock(q->results_mutex);
 				for (unsigned i = 0, n = results.size(); i < n; i++)
 				{
 					results[i]->position = q->allResults.size();
@@ -397,7 +410,7 @@ unsigned MinimizationQuery::loadResults(const MinimizationFilters& filter,
 			}
 		}
 		swap(results,tmpres);
-		
+
 	}
 	return total;
 }
